packages/components/src/autocomplete/autocompleter-ui.tsx

```diff
@@ -235,8 +232,7 @@ function useOnClickOutside(
 			document.removeEventListener( 'mousedown', listener );
 			document.removeEventListener( 'touchstart', listener );
 		};
-		// Disable reason: `ref` is a ref object and should not be included in a
+		// `ref` is a ref object and should not be included in a
```

Comment:
My worry is that, by having the disable comment, we will prevent future breakages of the rule from being surfaced by eslint.

In other words, if the disable comment is already there, I can make changes to the hook, forgetting to add the relevant dependencies, and ESLint won't tell me because the rule is already disabled. And the disable comment may be even more deceiving when debugging a potential issue, because it will potentially hide the real cause of the bug behind an "innocent" "refs are immutable" comment.

I think there are 2 better alternatives:

- ideally, we should be able to somehow mark `ref` as a react ref to eslint (or, somehow, to mark is as an immutable reference). But I don't think that's an option;
- that's why I think that a potentially better compromise would be to add `ref` to the list of dependencies (it shouldn't hurt in practical terms), which allows us to remove the disable comment. At most, we could leave a normal text comment mentioning that `ref` is there to satisfy lint rules, even if not necessary.

I would apply this reasoning in every place where it could allow us to remove the disable comment.

What do you think?

Reply:
I think you have a point here. Added the ref in 8fdcc76456e9d6c3a1cee4729555be4bce6d1734